packages/client/src/components/GrantNotes.vue

```diff
@@ -0,0 +1,270 @@
+    <div
+      v-if="loadMoreVisible"
+      class="px-3 mb-3"
+    >
+      <b-button
+        block
+        size="md"
+        variant="light"
+        class="show-more-btn"
+        data-test-show-more-btn
+        @click="fetchNextNotes"
+      >
+        Show More Notes
+      </b-button>
+    </div>
+  </section>
+</template>
+
+<script>
+import { nextTick } from 'vue';
+import { mapActions, mapGetters } from 'vuex';
+import UserAvatar from '@/components/UserAvatar.vue';
+import GrantNote from '@/components/GrantNote.vue';
+
+export default {
+  components: {
+    UserAvatar,
+    GrantNote,
+  },
+  emits: ['noteSaved'],
+  data() {
+    return {
+      otherNotes: [],
+      userNote: null,
+      noteText: '',
+      submittingNote: false,
+      editingNote: false,
+      notesNextCursor: null,
+    };
+  },
+  computed: {
+    ...mapGetters({
+      loggedInUser: 'users/loggedInUser',
+      currentGrant: 'grants/currentGrant',
+    }),
+    loadMoreVisible() {
+      return this.notesNextCursor !== null;
+    },
+    filteredNoteText() {
+      return this.noteText.trim();
+    },
+    emptyNoteText() {
+      return this.filteredNoteText.length === 0;
+    },
+    noteSendBtnDisabled() {
+      return this.emptyNoteText || this.submittingNote;
+    },
+    charCountClass() {
+      const errColor = this.filteredNoteText.length === 300 ? 'text-error' : '';
+
+      return `ml-auto ${errColor}`;
+    },
+    userNoteClass() {
+      const corners = this.emptyNoteText ? 'rounded-bottom-corners' : '';
+
+      return `user-note ${corners}`;
+    },
+  },
+  async beforeMount() {
+    this.fetchUsersNote();
+    this.fetchNextNotes();
+  },
+  methods: {
+    ...mapActions({
+      getNotesForGrant: 'grants/getNotesForGrant',
+      getNotesForCurrentUser: 'grants/getNotesForCurrentUser',
+      saveNoteForGrant: 'grants/saveNoteForGrant',
+    }),
+    formatter(value) {
+      return value.substring(0, 300);
+    },
+    async toggleEditNote() {
+      this.editingNote = true;
+      this.noteText = this.userNote.text;
+
+      await nextTick();
+      this.$refs.noteTextarea.focus();
+    },
+    handleKeyDown(e) {
+      if (e.key === 'Enter') {
+        this.$refs.submitNoteBtn.click();
+      }
+    },
+    async submitNote() {
+      this.submittingNote = true;
+
+      try {
+        await this.saveNoteForGrant({ grantId: this.currentGrant.grant_id, text: this.filteredNoteText });
+        this.$emit('noteSaved');
+        await this.fetchUsersNote();
+      } catch (e) {
+        // Error already logged
+      }
+
+      this.submittingNote = false;
+    },
+    async fetchUsersNote() {
+      const result = await this.getNotesForCurrentUser({ grantId: this.currentGrant.grant_id });
+
+      this.userNote = result && result.notes.length ? result.notes[0] : null;
+      this.editingNote = !this.userNote;
+    },
+    async fetchNextNotes() {
+      const query = {
+        grantId: this.currentGrant.grant_id,
+        limit: 4,
```

Comment:
Per a conversation I had with @ClaireValdivia recently about improving test-ability, what do you think of adding this function to `packages/client/src/helpers/featureFlags/index.js`, and calling that function on this line?

```js
export function grantNotesLimit() {
  return parseInt(getFeatureFlags().grantNotesLimit, 10) || 4;
}
```

This would enable users to run something like `window.APP_CONFIG.overrideFeatureFlag(grantNotesLimit, 1)` and change the pagination size on-the-fly.

Reply:
very useful idea - added!